### ISSUE TITLE
AWS A100 runners reliability issue

### ISSUE THREAD

**Issue by huydhn** [pytorch/pytorch#140332]
Compiler team reports that there are still some reliability issues with AWS A100 where some runners start to crash since last weekend. For example,

* https://github.com/pytorch/pytorch/actions/runs/11763098772
* https://github.com/pytorch/pytorch/actions/runs/11773760797/attempts/1

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @seemethere @malfet @pytorch/pytorch-dev-infra @jeanschmidt @ZainRizvi @desertfire 

Also @yangw-dev as this might be an interesting case for utilization monitoring because:

* Runner lost communication with server usually indicates OOM or a botched runner update.  Monitoring memory usage would help with OOM
* The hardware is shared, but each runner runs independently in its own cgroups / docker container https://fb.workplace.com/groups/pytorch.dev.perf.infra.teams/permalink/9137566856262424.  We probably want to monitor each of them independently

**Comment by malfet** [comment]
> Runner lost communication with server usually indicates OOM or a botched runner update. Monitoring memory usage would help with OOM

It creates a chicken-and-egg problem, as in case of OOM, monitoring process is likely the one that will be killed :)
But more fundamentally, it'll be really nice to replace/extend our AWS log-classifer lambda with something that would stream the logs to S3 bucket while job is running, so that in case runner dies will be able to recover some information

**Comment by jeanschmidt** [comment]
Definately, the next steps to mitigate this issue would be improving our monitoring tooling so we can identify what is happening to those instances. Given it is difficult to do a post-mortem analysis without any additional information.

I'll be working on that, and keep updating things here as they evolve.

**Comment by jeanschmidt** [comment]
Disabling the SPA to avoid interfering with the next jobs: https://github.com/pytorch-labs/pytorch-gha-infra/pull/530

**Comment by huydhn** [comment]
I found an example where the runner ran out of disk space here https://github.com/pytorch/pytorch/actions/runs/11876834273.  I guess this is the risk of delaying spa run when draining?

**Comment by huydhn** [comment]
Copy the note from the sync today with @jeanschmidt that we should have at least enough free space there for the process to dump all the memory to disk.  [https://aws.amazon.com/ec2/instance-types/p4/](https://www.google.com/url?q=https://aws.amazon.com/ec2/instance-types/p4/&sa=D&source=docs&ust=1732169071923655&usg=AOvVaw1UReNiscK5jsLU8g_Lo7Lq) has 1T+ of memory, so I think having a maybe 2T of disk space with some room to spare is good

**Comment by huydhn** [comment]
@jeanschmidt I'm seeing a 5TB partition available for the A100 runner, but only 113GB of home folder for each users there, which seems very low.  Is that correct? 

```
alice@ip-172-31-1-76:/var/snap/amazon-ssm-agent/7993$ df -h
Filesystem                      Size  Used Avail Use% Mounted on
/dev/root                       4.8T   73G  4.7T   2% /
tmpfs                           561G     0  561G   0% /dev/shm
tmpfs                           225G  1.6M  225G   1% /run
tmpfs                           5.0M     0  5.0M   0% /run/lock
/dev/nvme0n1p15                 105M  6.1M   99M   6% /boot/efi
/dev/mapper/vg.01-lv_ephemeral  6.8T   28K  6.5T   1% /opt/dlami/nvme
tmpfs                           113G  108K  113G   1% /run/user/1001
tmpfs                           113G  108K  113G   1% /run/user/1002
tmpfs                           113G  108K  113G   1% /run/user/1003
tmpfs                           113G  108K  113G   1% /run/user/1004
tmpfs                           113G  108K  113G   1% /run/user/1005
tmpfs                           113G  108K  113G   1% /run/user/1006
tmpfs                           113G  108K  113G   1% /run/user/1007
tmpfs                           113G  108K  113G   1% /run/user/1008
```

**Comment by huydhn** [comment]
I'm closing this now as the runner looks much stable now, will reopen / create new issues if needed

**Comment by huydhn** [comment]
Reported by @xmfan that the issue has come back since the past few days https://fb.workplace.com/groups/4571909969591489/permalink/8992516220864153/

**Comment by huydhn** [comment]
I'll mark this as high priority as this issue has been around for sometimes causing incomplete data on https://hud.pytorch.org/benchmark/compilers.  This issue has been reported by multiple folks (cc @ezyang, @zou3519, @shunting314, @desertfire)

I will work with @jeanschmidt looking for the root cause.  This is not a capacity issue as we have 30 A100 runners around atm.

**Comment by desertfire** [comment]
I think it's worth to try increasing the shard number in https://github.com/pytorch/pytorch/blob/main/.github/workflows/inductor-perf-test-nightly.yml since we have enough runners now. This way, each shard will run faster to finish and re-running a failed shard will also be acceptable until we fix the reliability issue.

**Comment by huydhn** [comment]
Yeah, let's increase the number of shards, there are spare capacity now.  Also @jeanschmidt has a potential fix here that will be rolled out soon https://github.com/pytorch-labs/pytorch-gha-infra/pull/597

**Comment by seemethere** [comment]
Going to close this since it seems like we've reached stability
